### ISSUE TITLE
Transform Anchor

### ISSUE THREAD

**Issue by DanielDK05** [bevyengine/bevy#12479]
## What problem does this solve or what need does it fill?

It would be nice to set an anchor point, like you can on `Sprite`, on either the transform or mesh.
I am currently creating a game where I would benefit from anchoring the transform on the bottom of the entity.

## What solution would you like?

I am envisioning a new component, like this:

```rust
enum TransformAnchor {
   Center,
   TopLeftFront,
   TopLeftCenter,
   TopLeftBack,
   TopRightFront,
   Custom(Vec3)
   etc...
}

commands.spawn((
   Transform::default(),
   TransformAnchor::BottomCenter,
));
```

## What alternative(s) have you considered?

An alternative would of course be to just offset the transform manually, but I find that a bit tedious to work with.

**Comment by KingBright** [comment]
I wonder if there is any method to rotate an image (or a sprite) on a specific anchor point?

**Comment by DanielDK05** [comment]
@copygirl I agree with this. I don't remember my exact thoughts when I made this issue, but a Vec3 wrapper seems like the most logical solution.
